Automattic/wp-calypso
# Fix/signup field validation

#### Changes proposed in this Pull Request

### The Problem

This PR attempts to remedy a couple of overly aggressive form validation use cases:

1) onBlur triggers validation on all fields (even one's we've yet to get to yet)

![](https://d1jfzjx68gj8xs.cloudfront.net/items/0O1h0G2S442u3Y113F1a/Screen%20Recording%202019-03-01%20at%2007.03%20AM.gif)

2) Debounce triggers validation on all fields (even one's we've yet to get to yet)

![](https://d1jfzjx68gj8xs.cloudfront.net/items/2T1A2e1e423S0N1V093E/Screen%20Recording%202019-03-01%20at%2001.21%20PM.gif)

### The Solution

With this patch applied, we fix the over aggressive validation messages:

![](https://d1jfzjx68gj8xs.cloudfront.net/items/3a3A040X3r0v1h0Z1115/Screen%20Recording%202019-03-01%20at%2001.26%20PM.gif)

#### Testing instructions

- Check out branch
- Recompile
- Test changes on http://calypso.localhost:3000/start/user

Fixes #16684
Which was technically closed in response to #18058, but that PR only fixed part of the problem (entering nothing and then clicking a link)
